src/server/external-services/pathway-commons.js

```diff
@@ -121,23 +125,70 @@ const sifGraph = async ( queryObj ) => {
   });
 };
 
-/* xref2Uri
- * Light wrapper around pc metadata service for mapping Xrefs db, id to url
+const handleEntityUriResponse = text => {
+  if( !text ) return ''; // Go ahead so we can cache, but gotta throw to inform clients
+
+  const uri = new url.URL( text ); // Throws TypeError
+  const pathParts = _.compact( uri.pathname.split('/') );
+  if( _.isEmpty( pathParts ) || pathParts.length !== 2 ) throw new Error( 'Unrecognized URI' );
+  const namespace = _.head( pathParts );
+  return uri.origin + '/' + namespace;
+};
+
+const constructQueryPath = ( name, localId ) => {
+  // Edge case - localId has periods e.g. 'enzyme nomenclature/6.1.1.5' gotta add a trailing slash
+  const suffix = /\./.test( localId ) ? '/' : '';
+  return name + '/' + localId + suffix;
+};
+
+/* fetchEntityUriBase
+ * Light wrapper around the pc2 service to get the uri given a collection name and local ID for entity
  * http://www.pathwaycommons.org/pc2/swagger-ui.html#!/metadata45controller/identifierOrgUriUsingGET
+ * NB: pc2 service returns 200 and empty body if collection name and/or local ID are unrecognized.
+ *   If the local ID is empty, throws a 404
+ * @return { object } a node URL object
+ */
+const fetchEntityUriBase = ( name, localId ) => {
+  //console.log( `fetchEntityUriBase with collection = ${name}; ID = ${localId}` );
+  const url = config.PC_URL + 'pc2/miriam/uri/' + constructQueryPath( name, localId ) ;
```

Comment:
@IgorRodchenkov  @jvwong does this return a result if I send the name 'uniprot knowledgebase' because that is one of the names that uniprot appears under in the jsonld biopax.


See: 

https://github.com/PathwayCommons/app-ui/blob/development/test/server/pathways/input.biopax.json#L564

Reply:
Nevermind, it does
http://www.pathwaycommons.org/pc2/miriam/uri/uniprot%20knowledgebase/P27695

I just got confused because Igor's comment here said that 'uniprot' maps to 'uniprot':

```
E.g., ‘uniprot’ maps to ‘uniprot’, ‘kegg compound’ - ‘kegg.compound’, ‘uniprot isoform’ - ‘uniprot.isoform’ (unfortunately, there are cases where simple replacing of a space with period and turning to lower case would be wrong)
```

Reply:
Indeed a bunch of possible collection names (e.g. UniProt Knowledgebase) map to the same namespace (e.g. uniprot) : MIRIAM 'preferred', MIRIAM 'synonym', PSI-MI controlled vocabulary 'database citation'. What rules they use for namespace is not clear.  E.g. 'Enzyme Nomenclature' maps to 'ec-code'...